pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
+	// Who currently has the lock.
+	HeldBy string `json:"heldby"`
+	// How long the lock is valid.
+	LeaseTime uint64 `json:"duration"`
```

Comment:
@wojtek-t I believe we can use util.Time for AcquiredTime and RenewTime, but LeaseTime is a duration at best and I don't see util.Time wrapping that functionality.  The number is plugged straight into the ttl for etcd, so even using a Duration adds unneeded extra processing.

Is that ok?